seesaw/seesaw_test.go

```diff
@@ -0,0 +1,84 @@
+package seesaw
+
+import (
+	"reflect"
+	"testing"
+	"time"
+)
+
+func TestDevice_SoftReset_success(t *testing.T) {
+
+	mocked := newMockDev(t, 0x49,
+		when([]byte{0x00, 0x7F, 0xFF}, nil, nil),
+		when([]byte{0x00, 0x01}, nil, nil),
+		when(nil, []byte{0x55}, nil),
+	)
+
+	sut := New(mocked)
+
+	err := sut.SoftReset()
+	assertDeepEquals(t, err, nil)
+}
+
+func TestDevice_WriteRegister(t *testing.T) {
+
+	write := byte(0x1F)
+	mocked := newMockDev(t, 0x49,
+		when([]byte{0x01, 0x04, write}, nil, nil),
+	)
+
+	sut := New(mocked)
+
+	err := sut.WriteRegister(ModuleGpioBase, FunctionGpioBulk, write)
+	assertDeepEquals(t, err, nil)
+}
+
+func TestDevice_ReadRegister(t *testing.T) {
+
+	read := byte(0x23)
+	mocked := newMockDev(t, 0x49,
+		when([]byte{0x0F, 0x10}, nil, nil),
+		when(nil, []byte{read}, nil),
+	)
+
+	sut := New(mocked)
+
+	r, err := sut.ReadRegister(ModuleTouchBase, FunctionTouchChannelOffset)
+	assertDeepEquals(t, err, nil)
+	assertDeepEquals(t, r, read)
+}
+
+func TestDevice_Read(t *testing.T) {
+
+	expectedRead := []byte{0x01, 0x02, 0x03, 0x04, 0x05}
+	mocked := newMockDev(t, 0x49,
+		when([]byte{0x0F, 0x10}, nil, nil),
+		when(nil, expectedRead, nil),
+	)
+
+	sut := New(mocked)
+
+	var buf [5]byte
+	err := sut.Read(ModuleTouchBase, FunctionTouchChannelOffset, buf[:], time.Nanosecond)
+	assertDeepEquals(t, err, nil)
+	assertDeepEquals(t, buf[:], expectedRead)
+}
+
+func TestDevice_Write(t *testing.T) {
+
+	expectedWrite := []byte{0x01, 0x02, 0x03, 0x04, 0x05}
+	mocked := newMockDev(t, 0x49,
+		when(append([]byte{0x0E, 0x04}, expectedWrite...), nil, nil),
+	)
+
+	sut := New(mocked)
+
+	err := sut.Write(ModuleNeoPixelBase, FunctionNeopixelBuf, expectedWrite)
+	assertDeepEquals(t, err, nil)
+}
+
+func assertDeepEquals[e any](t *testing.T, actual, expected e) {
```

Comment:
I tried quicktest again, does not compute for me. I.e. even the existing usages on `dev` don't work locally.

Reply:
>  even the existing usages on dev don't work locally.

That seems to imply something wrong with your local config. no?

Reply:
Very fair point :)

1. I now went through re-building tinygo from v0.30.0 and still get the same issue:
```
~> tinygo test -v ./tester
panic: unimplemented: (reflect.Type).IsVariadic()
FAIL    tinygo.org/x/drivers/tester     1.314s
~> tinygo version
tinygo version 0.30.0 linux/amd64 (using go version go1.20.8 and LLVM version 16.0.1)
```

2. then I had a look at the workflow on the repo that runs it successfully, if I do the same with the docker image I still get the error:
```
> docker run --rm -it -v /<redacted>/tinygo_drivers:/build ghcr.io/tinygo-org/tinygo-dev:latest /bin/bash
root@2632e6b72f70:/go# cd /build
root@2632e6b72f70:/build# tinygo test ./tester
go: downloading github.com/frankban/quicktest v1.10.2
...
go: downloading golang.org/x/xerrors v0.0.0-20191204190536-9bdfabe68543
panic: unimplemented: (reflect.Type).IsVariadic()
FAIL    tinygo.org/x/drivers/tester     1.366s
```

3. furthermore I found that even the [GitHub Action on the `dev`](https://github.com/tinygo-org/drivers/actions/runs/6516908120/job/17700893731) branch seems to succeed with failing tests:
![Screenshot from 2023-10-22 01-16-21](https://github.com/tinygo-org/drivers/assets/4774528/930708a8-b70d-4920-95d0-9b9200b0e5a0)

At this point I'm at a loss what the actual problem is and whether my environment is the problem or the CI environment has some issues.
I also made a PR to fix the tests #614

Any pointers? At least the docker environment should work, no?

Reply:
Ok, I figured it out :facepalm: 

Apparently the way to run unit tests is to use `go test` and not `tinygo test`.

Out of curiosity, what's the purpose of `tinygo test` then?

It runs tests fine without the `quicktest` dependency :D